src/main/java/pages/FlightSearchPage.java

```diff
@@ -0,0 +1,235 @@
+                } catch (Exception e) {
+                    return false;
+                }
+            });
+        }
+    }
+
+    public void selectDepartureCity(String cityName) {
+        fromButton.click();
+        wait.until(ExpectedConditions.presenceOfElementLocated(CITY_DROPDOWN_LIST));
+
+        By cityLocator = By.xpath(String.format(CITY_DROPDOWN_ITEM, cityName));
+        wait.until(ExpectedConditions.presenceOfElementLocated(cityLocator));
+        WebElement city = wait.until(ExpectedConditions.elementToBeClickable(cityLocator));
+
+        scrollIntoView(city);
+        city.click();
+    }
+
+    public void selectArrivalCity(String cityName) {
+        wait.until(ExpectedConditions.elementToBeClickable(toButton)).click();
+        try {
+            WebElement searchBox = wait.until(ExpectedConditions.presenceOfElementLocated(ARRIVAL_SEARCH_BOX));
+            searchBox.clear();
+            searchBox.sendKeys(cityName);
+            searchBox.sendKeys(Keys.ENTER);
+        } catch (TimeoutException e) {
+            WebElement city = wait.until(ExpectedConditions.elementToBeClickable(
+                    By.xpath(String.format(CITY_DROPDOWN_ITEM, cityName))));
+            scrollIntoView(city);
+            city.click();
+        }
+    }
+
+    public void selectDepartureDate(String day, String targetMonth) {
+        jsClick(departureInput);
+        wait.until(ExpectedConditions.visibilityOf(calendarLeft));
+
+        final int maxRetries = 12;
+        for (int i = 0; i < maxRetries; i++) {
+            if (isTargetMonthVisible(targetMonth)) {
+                break;
+            }
+            wait.until(ExpectedConditions.elementToBeClickable(nextButtonLeft)).click();
+            waitForCalendarUpdate();
+        }
+
+        if (!isTargetMonthVisible(targetMonth)) {
+            throw new NoSuchElementException("Target month not visible after max retries: " + targetMonth);
+        }
+
+        WebElement correctCalendar = getCalendarForMonth(targetMonth);
+        WebElement dateCell = correctCalendar.findElement(By.xpath(String.format(CALENDAR_DATE_CELL, day)));
+        scrollIntoView(dateCell);
+        dateCell.click();
+    }
+
+    public void selectReturnDate(String day) {
+        WebElement returnInput = getReturnInput();
+        jsClick(returnInput);
+        wait.until(ExpectedConditions.visibilityOf(calendarRight));
+        WebElement dateCell = calendarRight.findElement(By.xpath(String.format(CALENDAR_DATE_CELL, day)));
+        scrollIntoView(dateCell);
+        dateCell.click();
+    }
+
+    public void selectPassengers(int adults, int children, int babies) {
+        wait.until(ExpectedConditions.elementToBeClickable(passengersButton));
+        scrollIntoView(passengersButton);
+        jsClick(passengersButton);
+        wait.until(ExpectedConditions.visibilityOf(passengerPopover));
+
+        int currentAdults = Integer.parseInt(adultInput.getAttribute("value"));
```

Comment:
Make getAttribute("value") reusable, that might be used for other elements as well